### PR TITLE
feat: Add random generated suffix to COS bucket name

### PR DESCRIPTION
**Pull request by akocbek**
### Description

Adding optional feature which adds random generated suffix to a COS bucket.

#### No release required

- [ ] Examples or tests (addition or updates of examples or tests)
- [ ] Documentation update
- [ ] CI-related update (pipeline, etc.)
- [ ] Other changes that don't affect Terraform code

#### Release required

- [ ] Bug fix (patch release (`x.x.X`): Change that fixes an issue and is compatible with earlier versions)
- [x] New feature (minor release (`x.X.x`): Change that adds functionality and is compatible with earlier versions)
- [ ] Breaking change (major release (`X.x.x`): Change that is likely incompatible with previous versions)

##### Release notes content

Replace this text with information that users need to know about the bug fixes, features, and breaking changes. This information helps the merger write the commit message that is published in the release notes for the module.

---

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- Merge by using "Squash and merge".
- Use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author.

    The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).

### CODE REVIEW

**Review by ocofaigh — COMMENTED** [review]
The issue mentions `suffix / prefix`- do you think we need to support both?

**Review by shemau — CHANGES_REQUESTED** [review]
Why is this pushed into the main module? Problems with this approach:

- It adds a required provider, will all consumers have to include provider random now?
- It restricts the suffix to random and 4 characters, when the feature exists different consumers will have different requirements
- It adds a resource (the random string) that all consumers will see, including consumers that are not creating buckets, and consumers that set the flag `add_bucket_name_suffix` to false or let it default. This will lead to confusion
- Need to skip upgrade testing

Is a better approach to treat suffix like prefix and include it in the examples?

- No require providers, the consumer can decide if a provider is warranted or generate the string other ways
- The consumer can control the generation
- No new resources in the main module

If it is decided to keep this in the main module, then there should be a count on the new resource to prevent it showing is create_cos_buckets is false or add_bucket_name_suffix is false.

Also, how does the modified complete example work without defining a provider or version for hashicorp/random?

**Comment by ocofaigh** [comment]
@shemau we discussed earlier with @vburckhardt and it was confirmed this should be a main module (opt-in) feature

**Comment by akocbek** [comment]
@shemau thank you for your input. I added count on `random_string` resource, which doesn't get created in a case we do not want to add suffix to bucket name.

File: variables.tf [hunk 1]
```diff
@@ -108,6 +108,12 @@ variable "bucket_name" {
   default     = null
 }
 
+variable "add_bucket_name_suffix" {
+  type        = bool
+  description = "Add random generated suffix to the newly provisioned COS bucket name (Optional)."
```
Comment (ocofaigh) [inline variables.tf:113]: perhaps mention the suffix is 4 characters long

**Pull request merged**